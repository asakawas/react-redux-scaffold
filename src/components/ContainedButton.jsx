// @flow

import React from 'react';
import type { Node } from 'react';
import Button from '@material-ui/core/Button';
import PropTypes from 'prop-types';
import type { ActionCreator } from '../actions/actions';

type Props = {
  text: string,
  onClick: ActionCreator,
};

const ContainedButton = ({ text, onClick }: Props): Node => (
  <Button variant="contained" onClick={onClick}>
    {text}
  </Button>
);

ContainedButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default ContainedButton;
