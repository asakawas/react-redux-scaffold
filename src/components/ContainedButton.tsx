import React, { FunctionComponent } from 'react';
import Button from '@material-ui/core/Button';
import { zundokoButtonClicked } from '../actions/actions';

type Props = {
  text: string;
  onClick: typeof zundokoButtonClicked;
};

const ContainedButton: FunctionComponent<Props> = ({ text, onClick }) => (
  <Button variant="contained" onClick={onClick}>
    {text}
  </Button>
);

export default ContainedButton;
