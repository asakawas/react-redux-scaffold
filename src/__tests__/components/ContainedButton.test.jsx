import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import ContainedButton from '../../components/ContainedButton';

describe('components', () => {
  describe('ContainedButton', () => {
    test('renders correctly', () => {
      const { asFragment } = render(<ContainedButton text="ZUNDOKO" onClick={() => {}} />);
      expect(asFragment()).toMatchSnapshot();
    });

    test("calls the passed handler when it's clicked", () => {
      const handler = jest.fn();
      const { getByText } = render(<ContainedButton text="ZUNDOKO" onClick={handler} />);
      fireEvent.click(getByText('ZUNDOKO'));
      expect(handler).toHaveBeenCalledTimes(1);
    });
  });
});
