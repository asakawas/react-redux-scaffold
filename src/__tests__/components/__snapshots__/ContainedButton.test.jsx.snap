// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ContainedButton renders correctly 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label"
    >
      ZUNDOKO
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
